apps/dashboard/pkg/migration/schemaversion/v34.go

```diff
@@ -0,0 +1,294 @@
+// Before migration:
+//
+//	annotation: {
+//	  name: "CloudWatch Alerts",
+//	  dimensions: {"InstanceId": "i-123"},
+//	  namespace: "AWS/EC2",
+//	  region: "us-east-1",
+//	  prefixMatching: false,
+//	  statistics: ["Maximum", "Minimum"]
+//	}
+//
+// After migration:
+//
+//	annotations: [
+//	  { name: "CloudWatch Alerts - Maximum", dimensions: {"InstanceId": "i-123"}, namespace: "AWS/EC2", region: "us-east-1", prefixMatching: false, statistic: "Maximum" },
+//	  { name: "CloudWatch Alerts - Minimum", dimensions: {"InstanceId": "i-123"}, namespace: "AWS/EC2", region: "us-east-1", prefixMatching: false, statistic: "Minimum" }
+//	]
+func V34(dashboard map[string]interface{}) error {
+	dashboard["schemaVersion"] = int(34)
+
+	// Migrate panel queries if panels exist
+	panels, ok := dashboard["panels"].([]interface{})
+	if ok {
+		for _, panel := range panels {
+			p, ok := panel.(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			migrateCloudWatchQueriesInPanel(p)
+
+			// Handle nested panels in collapsed rows
+			if nestedPanels, hasNested := p["panels"].([]interface{}); hasNested {
+				for _, nestedPanel := range nestedPanels {
+					if np, ok := nestedPanel.(map[string]interface{}); ok {
+						migrateCloudWatchQueriesInPanel(np)
+					}
+				}
+			}
+		}
+	}
+
+	// Always migrate annotation queries regardless of whether panels exist
+	migrateCloudWatchAnnotationQueries(dashboard)
+
+	return nil
+}
+
+// migrateCloudWatchQueriesInPanel migrates CloudWatch queries within a panel that use multiple statistics.
+func migrateCloudWatchQueriesInPanel(panel map[string]interface{}) {
+	targets, ok := panel["targets"].([]interface{})
+	if !ok {
+		return
+	}
+
+	var newTargets []interface{}
+	var additionalTargets []interface{}
+
+	for _, target := range targets {
+		t, ok := target.(map[string]interface{})
+		if !ok {
+			newTargets = append(newTargets, target)
+			continue
+		}
+
+		if !isCloudWatchQuery(t) {
+			newTargets = append(newTargets, target)
+			continue
+		}
+
+		// Check if this query has multiple statistics
+		statistics, hasStatistics := t["statistics"].([]interface{})
+		if !hasStatistics || len(statistics) <= 1 {
+			// Convert single statistic or no statistics to proper format
+			if hasStatistics && len(statistics) == 1 {
+				if stat, ok := statistics[0].(string); ok {
+					t["statistic"] = stat
+				}
+				delete(t, "statistics")
+			}
+			newTargets = append(newTargets, t)
+			continue
+		}
+
+		// Split query with multiple statistics into separate queries
+		for i, stat := range statistics {
+			if statString, ok := stat.(string); ok {
+				// Create a copy of the original query
+				newQuery := make(map[string]interface{})
+				for k, v := range t {
+					if k != "statistics" {
+						newQuery[k] = v
+					}
+				}
+
+				// Set the single statistic
+				newQuery["statistic"] = statString
+
+				if i == 0 {
+					// First query replaces the original
+					newTargets = append(newTargets, newQuery)
```

Comment:
If the string conversion check above fails on i == 0, will that cause problems here?

Reply:
Very good catch! 🙌 

I have pushed some changes.
Now we first collect all valid string statistics, then migrate them. If no valid statistics exist, we still remove the statistics field and replace the original query/annotation. This will always replace the original query/annotation, guaranteeing the statistics field is removed regardless of which (or how many) statistics fail string conversion.

WDYT?